### PR TITLE
fix(dnd): place Move in the creative header overflow menu

### PR DESCRIPTION
**Pull request by sh1nj1**
Move the repeated row-level Move actions into the breadcrumb header’s overflow menu. The action uses selected creatives when present, otherwise the current creative; at the root with no selection it starts selection mode. Cancel and completion return focus to the visible overflow toggle.

Remove the actions from both trees and broadcast-rendered rows. Preserve read-only link creation and the shared move command. Production code decreases by 24 lines.

Validation: all 2,220 Jest tests and build pass; move controller coverage is 100% across statements, branches, functions, and lines. Vrex reviewed the change and ran the affected Rails helper, broadcast, and controller tests plus six browser system tests covering keyboard, click, cancellation, read-only linking, root selection, and multiple selection. Preview updated for visual review.

### CODE REVIEW

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 1]
```diff
@@ -26,13 +26,21 @@ export default class extends Controller {
...
+    this.ids = selected.length ? [...new Set(selected)] : id ? [id] : []
+    if (!this.ids.length) {
+      const select = document.getElementById('select-creative-btn')
+      if (select?.getAttribute('aria-pressed') !== 'true') select?.click()
+      document.querySelector('.select-creative-checkbox')?.focus()
+      return
+    }
+    this.canMove = selected.length
+      ? [...document.querySelectorAll('.select-creative-checkbox:checked')]
+        .every(el => el.closest('creative-tree-row')?.hasAttribute('can-write') === true)
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: @codex address this feedback in PR #1642. Preserve the previous exclusion of archived creatives for both Move and Link when opening the header action, including mixed active/archived selections. Check the actual row archived attribute representation, add regression coverage for archived-only and mixed selections, run the relevant tests, and push the fix to this PR. Report the commit and test results in this thread; leave resolution pending until the fix is verified.

[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: Implemented in e1882e9c4955f3b7e09cb50d5f4150b9735e7eef. creative-tree-row declares archived as a reflected Boolean attribute, so the guard uses hasAttribute('archived'), including archived="". Any selected archived row blocks the entire header action for both Move and Link, clears stale command state, and announces an EN/KO message asking the user to deselect archived creatives. Mixed selections are not silently partially processed. The existing helper exclusion for an archived current creative remains intact.

Regression coverage exercises archived-only and mixed selections, both modes, empty and explicit archived attributes, and reopening after deselection. Relevant Jest: 28/28 pass; controller statements/branches/functions/lines: 100%; npm run build and git diff --check pass. Published tree SHA matches the locally tested index. Ruby is unavailable in this environment, so Rails checks remain for CI. Leaving this thread unresolved pending verification.

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 2]
```diff
@@ -26,13 +26,29 @@ export default class extends Controller {
...
-    const selected = [...document.querySelectorAll('.select-creative-checkbox:checked')].map(el => el.value)
-    this.ids = selected.includes(id) ? [...new Set(selected)] : [id]
-    this.canMove = ![...document.querySelectorAll('[data-creative-move-writable="false"]')]
-      .some(el => this.ids.includes(el.dataset.creativeMoveId))
+    const selectedRows = [...document.querySelectorAll('.select-creative-checkbox:checked')]
+    // Lit reflects archived as a Boolean attribute, including archived="".
+    if (selectedRows.some(el => el.closest('creative-tree-row')?.hasAttribute('archived'))) {
+      this.ids = []
+      this.targetId = null
+      this.announcementTarget.textContent = this.messagesValue.archived
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: Implemented in d9ea3873fe1ba047001ba61131eb4ddeb1e4d85c. Archived-selection rejection now shows the existing EN/KO explanation in a visible native alert dialog, retains the live announcement, and restores trigger focus after dismissal. Archived-only and mixed selections still block both Move and Link without executing a command. Regression checks verify the visible alert and message for each case, focus restoration, and no alert after archived rows are deselected.

Relevant Jest: 28/28 pass; controller coverage: 100% statements, branches, functions, and lines. npm run build and git diff --check pass; published tree matches the tested index. Ruby is unavailable locally; Rails verification remains with CI. Leaving resolution pending verification.

File: engines/collavre/app/views/collavre/creatives/index.html.erb [hunk 3]
```diff
@@ -91,13 +91,15 @@
...
-            button_classes: 'creative-header-outline-btn'
+            button_classes: 'creative-header-outline-btn',
+            button_attributes: { aria: { controls: 'creative-overflow-menu', expanded: false } }
           ) do %>
         <% if (@parent_creative || @creative) && (@parent_creative || @creative).has_permission?(Current.user, :write) %>
           <button type="button" id="share-creative-btn" class="popup-menu-item" data-action="click->share-modal#open" data-share-modal-url-param="<%= collavre.creative_creative_shares_path(current_creative) %>">
             <span aria-hidden="true"><%= svg_tag 'share.svg', width: 16, height: 16 %></span> <%= t('collavre.creatives.index.share') %>
           </button>
         <% end %>
+        <%= render_creative_move_action(current_creative, current_creative&.has_permission?(Current.user, :write)) %>
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/views/collavre/creatives/index.html.erb]: Implemented in 139bfc5f15c3b7bc87166dfacf2dce5d51207cc7. The header capability now combines write permission with !current_creative.effective_origin.read_only_source?, matching the workspace tree and covering linked shells whose origin is externally managed. The existing controller defaults these sources to Link and disables Move; selected rows continue to use their existing can-write capability.

Added a request regression for a registered read-only source, its linked shell, and an ordinary writable creative, all with write permission. Strengthened the existing JavaScript test to assert that Move is disabled for the read-only header capability. Relevant Jest: 28/28 pass; git diff --check passes; published tree matches the locally tested index. Ruby is unavailable locally, so the new Rails regression is not yet executed here; requested Vrex's Mac review and Rails validation. Leaving resolution pending verification.

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 4]
```diff
@@ -26,13 +26,30 @@ export default class extends Controller {
...
+    const selected = selectedRows.map(el => el.value)
+    this.ids = selected.length ? [...new Set(selected)] : id ? [id] : []
+    if (!this.ids.length) {
+      const select = document.getElementById('select-creative-btn')
+      if (select?.getAttribute('aria-pressed') !== 'true') select?.click()
+      document.querySelector('.select-creative-checkbox')?.focus()
+      return
+    }
+    this.canMove = selected.length
+      ? selectedRows.every(el => el.closest('creative-tree-row')?.hasAttribute('can-write') === true)
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js:48]: Implemented in 97d81c9d30a1e0908544d5c3da2c6f4649f1a11c. TreeBuilder#can_write? now checks creative.effective_origin.read_only_source?, matching the header and workspace capability. Selected linked shells therefore receive can_write: false and the existing selection path defaults to Link and disables Move, even under a writable header.

Added Rails coverage for registered sources, their linked shells, ordinary writable sources and their shells, all with owner write permission. Added JavaScript regression coverage for selected read-only-origin shells alone and in mixed selections: Link default, disabled Move, forbidden Move submission blocked, and Link preserves shell IDs. Relevant Jest (move controller + tree renderer): 35/35 pass; git diff --check passes; published tree matches the tested local index. Ruby is unavailable locally; requested Vrex review and Mac Rails/lint validation. Leaving this thread unresolved pending verification.

File: engines/collavre/app/views/collavre/creatives/index.html.erb [hunk 5]
```diff
@@ -91,13 +91,15 @@
...
-            button_classes: 'creative-header-outline-btn'
+            button_classes: 'creative-header-outline-btn',
+            button_attributes: { aria: { controls: 'creative-overflow-menu', expanded: false } }
           ) do %>
         <% if (@parent_creative || @creative) && (@parent_creative || @creative).has_permission?(Current.user, :write) %>
           <button type="button" id="share-creative-btn" class="popup-menu-item" data-action="click->share-modal#open" data-share-modal-url-param="<%= collavre.creative_creative_shares_path(current_creative) %>">
             <span aria-hidden="true"><%= svg_tag 'share.svg', width: 16, height: 16 %></span> <%= t('collavre.creatives.index.share') %>
           </button>
         <% end %>
+        <%= render_creative_move_action(current_creative, current_creative&.has_permission?(Current.user, :write) && !current_creative.effective_origin.read_only_source?) %>
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/views/collavre/creatives/index.html.erb]: Implemented in 11ce5175e9edcbe07eb23896814261bb70658a29. The overflow Move launcher now renders only when a readable current creative exists or the request is the actual root (blank id). Inaccessible and missing requested IDs no longer fall back to the root launcher. Existing archived, authentication and read-only-source capability checks are preserved.

Added request regressions for inaccessible and missing IDs in both full-page and Turbo-frame requests, plus actual-root and blank-ID root availability. Existing readable-creative coverage remains. git diff --check passes and the published tree matches the local index. Rails execution was attempted but Ruby is unavailable here (/usr/bin/env: ‘ruby’: No such file or directory); requesting Mac Rails validation and review. Leaving resolution pending verification.

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 6]
```diff
@@ -26,13 +26,30 @@ export default class extends Controller {
...
+      window.alert(this.messagesValue.archived)
+      this.restoreFocus()
+      return
+    }
+    const selected = selectedRows.map(el => el.value)
+    this.ids = selected.length ? [...new Set(selected)] : id ? [id] : []
+    if (!this.ids.length) {
+      const select = document.getElementById('select-creative-btn')
+      if (select?.getAttribute('aria-pressed') !== 'true') select?.click()
+      document.querySelector('.select-creative-checkbox')?.focus()
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: Implemented in 5a48fdb4501f0aa46ef9e659da7c7d37d2dc2dc2. The root action now defers selection mode until CSR checkboxes exist, keeping focus on the visible overflow toggle (the Select item itself is also hidden inside that menu). On arrival it activates selection and focuses the first checkbox. The observer abandons the transition when focus has moved elsewhere and is disconnected on controller teardown. Newly connected checkbox targets also inherit the active selection mode without taking focus.

Regression coverage checks delayed arrival, focus moving away, disconnection, subsequent source selection, late checkbox visibility, and cancellation. Relevant Jest: 38/38 pass; npm run build and git diff --check pass. Published tree matches the tested local index. Full Jest is running; requested Vrex code review and Mac browser verification. Leaving resolution pending verification.

[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: Full JavaScript regression also passed on the published tree: 156 suites, 2,228 tests, zero failures. The fetched PR tree SHA is identical to the local tested index (6fea405df96a50ef793921fc6dcce619b97edc6f). CI and requested Mac browser review remain pending.

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 7]
```diff
@@ -45,6 +60,28 @@ export default class extends Controller {
...
+    }
+    // The selection toggle also lives in the closed overflow menu. Wait on
+    // its visible launcher until CSR supplies a source, without stealing focus.
+    this.restoreFocus()
+    this.focusObserver = new MutationObserver(() => {
+      if (document.activeElement !== this.trigger || !this.trigger.isConnected) {
+        this.focusObserver.disconnect()
+      } else if (document.querySelector('.select-creative-checkbox')) {
+        this.focusObserver.disconnect()
+        this.startSelection()
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: Implemented in 5d9145f722c49a07f50098026dc1baa2348e8149. Root selection now recognizes the tree controller's existing data-loaded completion flag, including attribute-only transitions. When loading finishes without selectable rows, it disconnects the observer and shows an EN/KO explanation through the shared in-app alertDialog, retaining the live announcement and restoring focus after dismissal. Already-completed empty views get the same feedback immediately. Abandoned requests remain cancelled when focus moves away or the controller disconnects; later rows cannot restart a completed wait.

Relevant Jest (move, select-mode, tree controllers): 83/83 pass. Regression coverage includes immediate/delayed empty completion, focus changes, teardown, later row insertion, and asynchronous dialog dismissal. npm run build and git diff --check pass; published tree matches the tested local index. Ruby is unavailable locally, so Rails/browser verification is requested from Vrex. Leaving this thread unresolved pending verification.

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 8]
```diff
@@ -45,6 +61,36 @@ export default class extends Controller {
...
 
+  startSelection() {
+    const checkbox = document.querySelector('.select-creative-checkbox')
+    if (checkbox) {
+      const select = document.getElementById('select-creative-btn')
+      if (select?.getAttribute('aria-pressed') !== 'true') select?.click()
+      checkbox.focus()
+      return
+    }
+    if (document.getElementById('creatives')?.dataset.loaded === 'true') {
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: Implemented in feef9d5fb0069f70980a16e1af4b9a61b50709ee. TreeController now publishes a separate data-load-state (success/error), clears it when a fresh request starts, and marks exhausted retries as error. The root action preserves the tree's localized load-error feedback through the in-app dialog rather than reporting an empty result. Successful empty results keep the existing empty explanation. Both completed and pending failures stop waiting without later row insertion reviving selection.

Relevant Jest: 87/87 pass (move, select-mode, tree controllers), including HTTP/JSON failures, exhausted retries, recovery to successful empty, immediate/delayed root failures, and existing empty-result cases. npm run build and git diff --check pass. Published tree matches the tested index (5ca344fa37ce1656109b5c700d3b3206fd829606). Ruby is unavailable locally; Mac browser/Rails review is requested. Leaving resolution pending verification.

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 9]
```diff
@@ -26,13 +27,28 @@ export default class extends Controller {
...
-    this.ids = selected.includes(id) ? [...new Set(selected)] : [id]
-    this.canMove = ![...document.querySelectorAll('[data-creative-move-writable="false"]')]
-      .some(el => this.ids.includes(el.dataset.creativeMoveId))
+    const selectedRows = [...document.querySelectorAll('.select-creative-checkbox:checked')]
+    // Lit reflects archived as a Boolean attribute, including archived="".
+    if (selectedRows.some(el => el.closest('creative-tree-row')?.hasAttribute('archived'))) {
+      this.ids = []
+      this.targetId = null
+      this.announcementTarget.textContent = this.messagesValue.archived
+      window.alert(this.messagesValue.archived)
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js]: Implemented in feef9d5fb0069f70980a16e1af4b9a61b50709ee. Archived-selection rejection now uses the imported shared alertDialog, with the existing EN/KO message and live announcement. Focus is restored after the promise settles, unless the controller disconnected. Empty/error feedback shares the same helper. Archived-only and mixed selections still block both Move and Link without partial execution.

Relevant Jest: 87/87 pass. Archived regression coverage retains both Boolean attribute representations and both modes, and verifies async dismissal/focus restoration plus teardown. npm run build and git diff --check pass; published tree matches the tested local index. Requested Mac desktop/browser review; leaving resolution pending verification.

File: engines/collavre/app/views/collavre/creatives/index.html.erb [hunk 10]
```diff
@@ -91,13 +91,17 @@
...
+            button_classes: 'creative-header-outline-btn',
+            button_attributes: { aria: { controls: 'creative-overflow-menu', expanded: false } }
           ) do %>
         <% if (@parent_creative || @creative) && (@parent_creative || @creative).has_permission?(Current.user, :write) %>
           <button type="button" id="share-creative-btn" class="popup-menu-item" data-action="click->share-modal#open" data-share-modal-url-param="<%= collavre.creative_creative_shares_path(current_creative) %>">
             <span aria-hidden="true"><%= svg_tag 'share.svg', width: 16, height: 16 %></span> <%= t('collavre.creatives.index.share') %>
           </button>
         <% end %>
+        <% if current_creative || params[:id].blank? %>
+          <%= render_creative_move_action(current_creative, current_creative&.has_permission?(Current.user, :write) && !current_creative.effective_origin.read_only_source?) %>
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/views/collavre/creatives/index.html.erb:103]: Implemented in b86df975b50a6e943de70829347799f692a942e0. The helper now retains the authenticated header launcher for archived current creatives but omits their source ID. With no selection it uses the existing selection flow; selected active children retain both Move and Link. Archived-only or mixed archived selections remain rejected by the existing guard, and signed-out/inaccessible views retain their exclusions.

Added helper and full-page/Turbo request regressions for an archived parent with an active child, plus JavaScript coverage for selection entry and exact child-only Move/Link command IDs. Relevant Jest: 47/47 pass; npm run build and git diff --check pass. Published tree matches the tested index (585c31170e4153ba28f317d57a40ce8a45242df8). Ruby is unavailable locally, so Rails/lint execution requires Mac/CI validation; requesting Vrex review. Leaving resolution pending verification.

File: engines/collavre/app/javascript/controllers/creative_move_controller.js [hunk 11]
```diff
@@ -26,13 +27,26 @@ export default class extends Controller {
...
+    // Lit reflects archived as a Boolean attribute, including archived="".
+    if (selectedRows.some(el => el.closest('creative-tree-row')?.hasAttribute('archived'))) {
+      this.ids = []
+      this.targetId = null
+      this.showFeedback(this.messagesValue.archived)
+      return
+    }
+    const selected = selectedRows.map(el => el.value)
+    this.ids = selected.length ? [...new Set(selected)] : id ? [id] : []
+    if (!this.ids.length) {
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/javascript/controllers/creative_move_controller.js:43]: Implemented in 897d56bd3d44860add4290c1f13bfc340d27bf35. SelectModeController now clears checked inputs, selected row styling, select-all and active/dragging state on reconnect and before Turbo caches the page, restoring the inactive controls and toggle label/aria-pressed. Row listener bookkeeping is initialized before target connection so restored rows can connect safely.

Regression coverage exercises cloned snapshots and same-element reconnects with stale checked inputs, confirms both Move and Link submit only the current creative ID, and verifies before-cache cleanup plus fresh selection afterward. Full Jest: 156 suites, 2,247 tests pass. npm run build and git diff --check pass. Published tree f3b5a35f66c2c6aafa649137ac0a4b4820120705 matches the locally tested tree. Ruby is unavailable here, so Rails/RuboCop/complexity execution could not run; requesting Mac review and real-browser Turbo history verification. Leaving resolution pending verification.
